compiler-cli/src/hex/auth.rs

```diff
@@ -173,6 +176,9 @@ fn ask_password(warnings: &mut Vec<Warning>) -> std::result::Result<String, Erro
         .inspect(|_| {
             warnings.push(Warning::DeprecatedEnvironmentVariable {
                 name: PASS_ENV_NAME.into(),
+                message: Some(format!(
+                    "Use the `{API_ENV_NAME}` environment variable instead."
+                )),
```

Comment:
Error messages should only ever be constructed in the error module, error and warning values we control should never hold text information themselves. This could be an enum indicating which deprecated environment variable it is